### PR TITLE
test: update unit tests for Pytorch flexible primitives [DET-3200]

### PR DESCRIPTION
**Pull request by shiyuann**
## Description

1. Update all the Pytorch unit tests with the new interface.
2. Merge all unit tests for checkpoint and restoring.
3. Merge the unit tests for LR schedulers.

## Test Plan

N/A

## Commentary (optional)

N/A

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: harness/determined/util.py [hunk 1]
```diff
@@ -26,7 +26,9 @@ def download_gcs_blob_with_backoff(blob: Any, n_retries: int = 32, max_backoff:
 
 
 def is_overridden(full_method: Any, parent_class: Any) -> bool:
-    return cast(bool, full_method.__qualname__.partition(".")[0] != parent_class.__name__)
+    if callable(full_method):
```
Comment (rb-determined-ai) [inline harness/determined/util.py:29]: question: What does this change do?

Comment (shiyuann) [reply harness/determined/util.py:29]: This is fix for that users can't set their optimizer by `self.optimizer=...`. We check if `self.optimizer` is overridden by checking `__qualname__` however, only functions have that attribute.

Comment (rb-determined-ai) [reply harness/determined/util.py:29]: Ah.  That's tricky.  We should explain that in a comment at a minimum.

Comment (shiyuann) [reply harness/determined/util.py:29]: Done

**Pull request merged**